backend/inspirehep/matcher/config.py

```diff
@@ -277,15 +282,93 @@
         {
             "queries": [
                 {"path": "ids.value", "search_path": "ids.value", "type": "exact"},
+            ],
+            "validator": authors_validator,
+        },
+        {
+            "queries": [
                 {
                     "path": "emails",
                     "search_path": "email_addresses.value",
                     "type": "exact",
                 },
-            ]
-        }
+            ],
+        },
     ],
     "index": "records-authors",
-    "source": ["control_number"],
+    "source": ["self.$ref"],
     "validator": "inspirehep.matcher.validators:authors_validator",
 }
+
+AUTHOR_MATCHER_NAME_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["first_name", "last_name"],
+                    "search_paths": ["authors.first_name", "authors.last_name"],
+                    "type": "nested",
+                    "inner_hits": {"_source": ["authors.record"]},
+                },
+            ],
+        },
+    ],
+    "index": "records-hep",
+    "source": ["control_number", "authors.full_name", "authors.record.$ref"],
+}
+
+
+AUTHOR_MATCHER_NAME_INITIALS_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "type": "author-names",
+                    "inner_hits": {"_source": ["authors.record"]},
+                },
+            ]
+        },
+    ],
+    "index": "records-hep",
+    "source": ["control_number", "authors.full_name", "authors.record.$ref"],
+}
+
+
+AUTHOR_MATCHER_COLLABORATION_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["first_name", "last_name"],
+                    "search_paths": ["authors.first_name", "authors.last_name"],
+                    "type": "nested",
+                    "inner_hits": {"_source": ["authors.record"]},
+                },
+            ],
+            "validator": collaboration_validator,
+        },
+    ],
+    "index": "records-hep",
+    "source": ["authors.record.$ref", "collaborations.value",],
+}
+
+
+AUTHOR_MATCHER_AFFILIATION_CONFIG = {
+    "algorithm": [
+        {
+            "queries": [
+                {
+                    "paths": ["first_name", "last_name"],
+                    "search_paths": ["authors.first_name", "authors.last_name"],
+                    "type": "nested",
+                    "inner_hits": {
+                        "_source": ["authors.record", "authors.affiliations.value"]
+                    },
+                },
+            ],
+            "validator": affiliations_validator,
```

Comment:
why do this with a validator, can't you pass the affiliations in the search?

Reply:
That will require another changes in matcher, because if I'll add affiliations to nested query it will pass affiliations as the value to the query (as an array) and that's wrong. If I'll pass affiliations to exact match I can't use `inner_hits`, so I'll have to go through all the authors in the result to find matched one.

Reply:
why is it wrong to pass affiliations to the nested query?

Reply:
Currently what's happening is:
```
{
        'query': {
            'nested': {
                'path': common_path,
                'query': {
                    'bool': {
                        'must': [
                         {"match": {
                        "authors.affiliations.value": ["CERN", "Warsaw U.]
                        }}
                       ],
                    },
                },
            },
        },
    }
```
So that would fail because of the array as the value. We can modify matcher once again to add a match query for each value if an array is passed if we can't leave the validator here.